homeassistant/components/rainbird/config_flow.py

```diff
@@ -0,0 +1,164 @@
+        )
+
+    async def _test_connection(self, host: str, password: str) -> str:
+        """Test the connection and return the device serial number.
+
+        Raises a ConfigFlowError on failure.
+        """
+        controller = AsyncRainbirdController(
+            AsyncRainbirdClient(
+                async_get_clientsession(self.hass),
+                host,
+                password,
+            )
+        )
+        try:
+            async with async_timeout.timeout(TIMEOUT_SECONDS):
+                return await controller.get_serial_number()
+        except asyncio.TimeoutError as err:
+            raise ConfigFlowError(
+                f"Timeout connecting to Rain Bird controller: {str(err)}",
+                "timeout_connect",
+            ) from err
+        except RainbirdApiException as err:
+            raise ConfigFlowError(
+                f"Error connecting to Rain Bird controller: {str(err)}",
+                "cannot_connect",
+            ) from err
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_HOST] == entry.data[CONF_HOST]:
+                return self.async_abort(reason="already_configured")
+        try:
+            serial_number = await self._test_connection(
+                config[CONF_HOST], config[CONF_PASSWORD]
+            )
+        except ConfigFlowError as err:
+            _LOGGER.error("Error during config import: %s", err)
+            return self.async_abort(reason=err.error_code)
+        return await self.async_finish(serial_number, config)
+
+    async def async_finish(
+        self,
+        serial_number: str,
+        data: dict[str, Any],
+        options: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Create the config entry."""
+        await self.async_set_unique_id(serial_number)
+        self._abort_if_unique_id_configured()
+        return self.async_create_entry(
+            title=data[CONF_HOST],
+            data=data,
+            options=options,
+        )
+
+
+class RainBirdOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a RainBird options flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize RainBirdOptionsFlowHandler."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        ATTR_DURATION,
+                        default=self.config_entry.options.get(
+                            ATTR_DURATION,
```

Comment:
Should we have a default duration for new users? If we set default options when we finish the config flow we don't need to have a default here.

Reply:
Updated to have default options instead.